packages/api-graphql/src/internals/APIClient.ts

```diff
@@ -778,3 +781,24 @@ export function authModeParams(
 		authToken: options.authToken || client[__authToken],
 	};
 }
+
+/**
+ * Retrieves custom headers from either the client or request options.
+ * @param {client} V6Client - for extracting client headers
+ * @param {requestHeaders} [CustomHeaders] - request headers
+ * @returns {CustomHeaders} - custom headers
+ */
+export function getCustomHeaders(
+	client: V6Client,
```

Comment:
Looking at the build error, this should probably be:
```suggestion
	client: V6Client | ClientWithModels,
```

Reply:
Just saw your comment now - already made this update locally (making a few additional updates now that strict mode is applied)